### PR TITLE
Update dec_lk201.cpp

### PR DESCRIPTION
**Pull request by Bavarese**
Prevent keyboard stuck errors by waiting for master reset (0xfd). Improve LK201 audio response (especially on autorepeat).

### CODE REVIEW

File: src/mame/machine/dec_lk201.cpp [hunk 1]
```diff
@@ -549,25 +553,34 @@ void lk201_device::device_timer(emu_timer &timer, device_timer_id id, int param,
...
+		if (m_beep_state == 2)
+		{
+			m_beeper->adjust(attotime::never);
+
+			m_speaker->set_state(0);
+			m_beep_state = 0;  // ready for new beep.
+
+		} else
+			m_beeper->adjust(attotime::from_msec(20));
+
```
Comment (cuavas) [inline src/mame/machine/dec_lk201.cpp]: Please make the use of line breaks consistent - if you're not putting opening brace on the same line as flow control statement, don't put the closing brace on the same line as a flow control statement, either.

File: src/mame/machine/dec_lk201.cpp [hunk 2]
```diff
@@ -549,25 +553,34 @@ void lk201_device::device_timer(emu_timer &timer, device_timer_id id, int param,
...
 	update_interrupts();
 	receive_register_extract();
 
-	int data = get_received_char();
-	m_kbd_state = data;
-//  printf("\nlk201 got %02x\n", m_kbd_state);
+	m_kbd_state = get_received_char();
+//	printf("\nlk201 got %02x\n", m_kbd_state);
+
+	if(m_kbd_state == 0xfd)
```
Comment (cuavas) [inline src/mame/machine/dec_lk201.cpp]: Please be consistent with spacing.  You have no space between if and the opening parenthesis, while you have a space there in other places in this file.

**Comment by cuavas** [comment]
This can't be merged until you push the corresponding changes to the header file into this pull request's branch.

**Comment by cracyc** [comment]
> Prevent keyboard stuck errors by waiting for master reset (0xfd). 

Not a big fan of this.  There are many users of the lk201 other than the rainbow and they may look for keypresses without doing a reset.  Plus the reset is handled by the mcu and if there is a stuck key we should find out why rather than adding logic that doesn't actually exist on the machine.

**Comment by Bavarese** [comment]
Very likely the key scan logic isn't complete, or keys dont fire at startup. We dont have schematics for the 68HC05. 
The patch was intended as a general workaround, as the keyboard must be reset at startup.
This is not a Rainbow specific issue (although very annoying there).

Should i close this one and wait until someone comes up with a better solution?

**Pull request closed without merging**